docs/concepts/pdp-sharding.mdx

```diff
@@ -0,0 +1,212 @@
+          socket_address: { address: 0.0.0.0, port_value: 80 }
+        filter_chains:
+        - filters:
+          - name: envoy.filters.network.http_connection_manager
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+              stat_prefix: ingress_http
+              codec_type: AUTO
+              route_config:
+                name: local_route
+                virtual_hosts:
+                - name: local_service
+                  domains: ["*"]
+                  routes:
+                  - match: { prefix: "/allowed" }
+                    route:
+                      cluster: pdp_cluster
+                      hash_policy:
+                        - header:
+                            header_name: X-Tenant-Id
+              http_filters:
+              - name: envoy.filters.http.router
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
+      clusters:
+      - name: pdp_cluster
+        connect_timeout: 0.25s
+        type: STRICT_DNS
+        lb_policy: RING_HASH
+        dns_lookup_family: V4_ONLY
+        load_assignment:
+          cluster_name: pdp_cluster
+          endpoints:
+          - lb_endpoints:
+         # Replicate for each Shard / PDP {
+            - endpoint:
+                address:
+                  socket_address:
+                    address: $PDP-1-ADDRESS
+                    port_value: 80
+              metadata:
+                filter_metadata:
+                  envoy.lb:
+                    hash_key: 1
+         # }. And another one as an example:
+            - endpoint:
+                address:
+                  socket_address:
+                    address: $PDP-2-ADDRESS
+                    port_value: 80
+              metadata:
+                filter_metadata:
+                  envoy.lb:
+                    hash_key: 2
+    ```
+3. Run Envoy:
+    ```sh
+   docker run --rm -it --network=envoy \
+      -v $(pwd)/my-custom-envoy.yaml:/envoy-custom.yaml \
+      -p 9901:9901 \
+      -p 10000:10000 \
+      envoyproxy/envoy:dev-d266941360e811eb34fc632d914211437e13f962 \
+          -c /envoy-custom.yaml
+    ```
+4. Run PDPs:
+    Each PDP should get its shard id as an environment variable:
+    ```sh
+    docker run -it \
+      -p 7766:7000 \
+      --name pdp-0 \
+      --network=envoy \
+      --env PDP_API_KEY=<YOUR_API_KEY> \
+      --env PDP_SHARD_ORD=0 \
+      permitio/pdp-v2:latest
+    ```
+    and
+    ```sh
+    docker run -it \
+      -p 7766:7000 \
+      --name pdp-1 \
+      --network=envoy \
+      --env PDP_API_KEY=<YOUR_API_KEY> \
+      --env PDP_SHARD_ORD=1 \
+      permitio/pdp-v2:latest
+   ```
+
+Finally you can send an authorization check request to the Envoy container:
+
+<Tabs groupId="language">
+<TabItem value="curl" label="cURL">
+
+```bash
+curl http://localhost:10000/allowed \
```

Comment:
Why is that needed?

Reply:
to show the actual permit.check

Reply:
Yes but I mean why the curl is needed? isn't the sdk enough?